overlord/configstate/configcore/handlers.go

```diff
@@ -0,0 +1,143 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/release"
+)
+
+type configHandler interface {
+	validate(config.ConfGetter) error
+	handle(config.ConfGetter, *fsOnlyContext) error
+	needsState() bool
+	flags() flags
+}
+
+// flags carries extra flags that influence how the handler is called.
+type flags struct {
+	// coreOnlyConfig tells Run/FilesystemOnlyApply to apply the config on core systems only.
+	coreOnlyConfig bool
+}
+
+type fsOnlyHandler struct {
+	validateFunc func(config.ConfGetter) error
+	handleFunc   func(config.ConfGetter, *fsOnlyContext) error
+	configFlags  flags
+}
+
+var handlers []configHandler
+
+func init() {
+	// Most of these handlers are no-op on classic.
+	// TODO: consider allowing some of these on classic too?
+	// consider erroring on core-only options on classic?
+
+	coreOnly := &flags{coreOnlyConfig: true}
+
+	// watchdog.{runtime-timeout,shutdown-timeout}
+	addFSOnlyHandler(validateWatchdogOptions, handleWatchdogConfiguration, coreOnly)
+
+	// Export experimental.* flags to a place easily accessible from snapd helpers.
+	addFSOnlyHandler(validateExperimentalSettings, doExportExperimentalFlags, nil)
+
+	// network.disable-ipv6
+	addFSOnlyHandler(validateNetworkSettings, handleNetworkConfiguration, coreOnly)
+
+	// service.*.disable
+	addFSOnlyHandler(nil, handleServiceDisableConfiguration, coreOnly)
+
+	// system.power-key-action
+	addFSOnlyHandler(nil, handlePowerButtonConfiguration, coreOnly)
+
+	// pi-config.*
+	addFSOnlyHandler(nil, handlePiConfiguration, coreOnly)
+}
+
+// addFSOnlyHandler registers functions to validate and handle a subset of
+// system config options that do not require to manipulate state but only
+// the file system.
+func addFSOnlyHandler(validate func(config.ConfGetter) error, handle func(config.ConfGetter, *fsOnlyContext) error, flags *flags) {
+	h := &fsOnlyHandler{
+		validateFunc: validate,
+		handleFunc:   handle,
```

Comment:
it does not make much sense to setup a fs-only handler that only validates, something that only validates usually relates to some internal state anyway.  This is true at moment,  so for now let's panic here if handleFunc is nil, then we don't need the if in handle() for fsOnyHandler.

Reply:
Done.